packages/@ember/-internals/glimmer/lib/helpers/action.ts

```diff
@@ -389,34 +389,32 @@ function makeDynamicClosureAction(
 }
 
 interface MaybeActionHandler {
-  actions?: Record<string, Function>;
+  actions?: Record<string, (...args: any[]) => any>;
 }
 
 function makeClosureAction(
   context: object,
   target: MaybeActionHandler,
-  action: string | Function,
+  action: string | ((...args: any[]) => any),
   processArgs: (args: unknown[]) => unknown[],
   debugKey: string
 ) {
   let self: object;
-  let fn: Function;
+  let fn: (...args: any[]) => any;
 
   assert(
     `Action passed is null or undefined in (action) from ${target}.`,
     action !== undefined && action !== null
   );
 
-  let typeofAction = typeof action;
-
-  if (typeofAction === 'string') {
+  if (typeof action === 'string') {
```

Comment:
Interesting. I assume it didn't type-check without this change? From TS 4.5 forward, I would expect it *would*. 🤔

Reply:
Yeah, it did not work :/

Reply:
Ahhhh, I see why. It isn't discriminating a union or anything quite like that, so the compiler loses track of it. 👍🏼